group/assignment6/RearrangeCars.cpp

```diff
@@ -0,0 +1,56 @@
+#include"RearrangeCars.h"
+
+RearrangeCars::RearrangeCars(const vector<int> &startStates) : startStates(startStates){}
+
+void RearrangeCars::setStartStates(const vector<int> &startStates){
+	this->startStates = startStates;
+}
+
+int RearrangeCars::findEmptySlotId(){
+	for (int slotId = 0; slotId < this->startStates.size(); slotId++){
+		if (this->startStates[slotId] == -1) return slotId; // no car has -1 id, it is reserved for empty slot
+	}
+	return -1; // This code should not be reached
+}
+
+vector<int> RearrangeCars::findCarPositions(const vector<int> &currentStates){
+	vector<int> carPositions(currentStates.size() - 1);
+	for (int i = 0; i < currentStates.size(); i++)
+	{

```

Comment:
You always keep { on the same lane as for, let's follow this style. Btw {} are not really necessary for this for loop (same in some other places in the code, we can remove them).